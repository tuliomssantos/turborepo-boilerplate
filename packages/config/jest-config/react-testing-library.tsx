import React from 'react'

import { render } from '@testing-library/react'

const customRender = (ui: React.ReactNode, options = {}) =>
  render(<React.Fragment>{ui}</React.Fragment>, {
    wrapper: ({ children }) => children,
    ...options,
  })

export * from '@testing-library/react'
export { customRender as render }
export { default as userEvent } from '@testing-library/user-event'
