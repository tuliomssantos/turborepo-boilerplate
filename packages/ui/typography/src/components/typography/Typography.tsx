import objStr from 'obj-str'

import styles from './styles.module.scss'

export type TypographyVariant =
  | 'display-2xl'
  | 'display-xl'
  | 'display-lg'
  | 'display-md'
  | 'display-sm'
  | 'display-xs'
  | 'text-xl'
  | 'text-lg'
  | 'text-md'
  | 'text-sm'
  | 'text-xs'

export type TypographyWeight = 'regular' | 'medium' | 'semi-bold' | 'bold'

export type TypographyProps<C extends React.ElementType> = {
  as?: C
  children: React.ReactNode
  variant?: TypographyVariant
  weight?: TypographyWeight
} & React.ComponentPropsWithoutRef<C>

const Typography = <C extends React.ElementType = 'span'>({
  as,
  children,
  variant = 'text-md',
  weight = 'medium',
  ...rest
}: TypographyProps<C>) => {
  const Component = as || 'span'

  return (
    <Component
      {...rest}
      //only set rest.className if it exists
      className={objStr({
        [styles.typography]: true,
        [styles[`typography__${variant}`]]: true,
        [styles[`typography__${weight}`]]: true,
        [rest?.className]: rest?.className,
      })}
    >
      {children}
    </Component>
  )
}

export default Typography
