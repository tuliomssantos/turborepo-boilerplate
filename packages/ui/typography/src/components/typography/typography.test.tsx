import { render, screen } from 'jest-common/react-testing-library'

import Typography from './Typography'

describe('Typography', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <Typography>
        Lorem ipsum, dolor sit amet consectetur adipisicing elit. Laborum
        veritatis soluta blanditiis ut, mollitia nisi ex architecto velit,
        expedita, iure illum quae perferendis? Illum beatae minima quo
        perferendis vero dolorum.
      </Typography>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it.each([
    'display-2xl',
    'display-xl',
    'display-lg',
    'display-md',
    'display-sm',
    'display-xs',
    'text-xl',
    'text-lg',
    'text-md',
    'text-sm',
    'text-xs',
  ])('should render correctly with variant %s', variant => {
    const { asFragment } = render(
      // @ts-expect-error - we're testing all variants
      <Typography variant={variant}>
        Lorem ipsum, dolor sit amet consectetur adipisicing elit. Laborum
        veritatis soluta blanditiis ut, mollitia nisi ex architecto velit,
        expedita, iure illum quae perferendis? Illum beatae minima quo
        perferendis vero dolorum.
      </Typography>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it.each(['regular', 'medium', 'semi-bold', 'bold'])(
    'should render correctly with weight %s',
    weight => {
      const { asFragment } = render(
        // @ts-expect-error - we're testing all weights
        <Typography weight={weight}>
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Laborum
          veritatis soluta blanditiis ut, mollitia nisi ex architecto velit,
          expedita, iure illum quae perferendis? Illum beatae minima quo
          perferendis vero dolorum.
        </Typography>
      )
      expect(asFragment()).toMatchSnapshot()
    }
  )

  it('should render correctly with className', () => {
    render(
      <Typography className="test-class">
        Lorem ipsum, dolor sit amet consectetur adipisicing elit. Laborum
        veritatis soluta blanditiis ut, mollitia nisi ex architecto velit,
        expedita, iure illum quae perferendis? Illum beatae minima quo
        perferendis vero dolorum.
      </Typography>
    )

    const typography = screen.getByText(
      'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Laborum veritatis soluta blanditiis ut, mollitia nisi ex architecto velit, expedita, iure illum quae perferendis? Illum beatae minima quo perferendis vero dolorum.'
    )

    expect(typography).toHaveClass('test-class')
  })

  it('should render correctly with as prop', () => {
    render(
      <Typography as="p">
        Lorem ipsum, dolor sit amet consectetur adipisicing elit. Laborum
        veritatis soluta blanditiis ut, mollitia nisi ex architecto velit,
        expedita, iure illum quae perferendis? Illum beatae minima quo
        perferendis vero dolorum.
      </Typography>
    )

    //test if informed as prop is being applied
    const typography = screen.getByText(
      'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Laborum veritatis soluta blanditiis ut, mollitia nisi ex architecto velit, expedita, iure illum quae perferendis? Illum beatae minima quo perferendis vero dolorum.'
    )

    expect(typography.tagName).toBe('P')
  })
})
