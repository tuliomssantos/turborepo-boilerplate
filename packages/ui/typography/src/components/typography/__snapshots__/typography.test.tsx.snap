// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography should render correctly 1`] = `
<DocumentFragment>
  <span
    class="typography typography__text-md typography__medium"
  >
    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Laborum veritatis soluta blanditiis ut, mollitia nisi ex architecto velit, expedita, iure illum quae perferendis? Illum beatae minima quo perferendis vero dolorum.
  </span>
</DocumentFragment>
`;

exports[`Typography should render correctly with variant display-2xl 1`] = `
<DocumentFragment>
  <span
    class="typography typography__display-2xl typography__medium"
  >
    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Laborum veritatis soluta blanditiis ut, mollitia nisi ex architecto velit, expedita, iure illum quae perferendis? Illum beatae minima quo perferendis vero dolorum.
  </span>
</DocumentFragment>
`;

exports[`Typography should render correctly with variant display-lg 1`] = `
<DocumentFragment>
  <span
    class="typography typography__display-lg typography__medium"
  >
    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Laborum veritatis soluta blanditiis ut, mollitia nisi ex architecto velit, expedita, iure illum quae perferendis? Illum beatae minima quo perferendis vero dolorum.
  </span>
</DocumentFragment>
`;

exports[`Typography should render correctly with variant display-md 1`] = `
<DocumentFragment>
  <span
    class="typography typography__display-md typography__medium"
  >
    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Laborum veritatis soluta blanditiis ut, mollitia nisi ex architecto velit, expedita, iure illum quae perferendis? Illum beatae minima quo perferendis vero dolorum.
  </span>
</DocumentFragment>
`;

exports[`Typography should render correctly with variant display-sm 1`] = `
<DocumentFragment>
  <span
    class="typography typography__display-sm typography__medium"
  >
    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Laborum veritatis soluta blanditiis ut, mollitia nisi ex architecto velit, expedita, iure illum quae perferendis? Illum beatae minima quo perferendis vero dolorum.
  </span>
</DocumentFragment>
`;

exports[`Typography should render correctly with variant display-xl 1`] = `
<DocumentFragment>
  <span
    class="typography typography__display-xl typography__medium"
  >
    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Laborum veritatis soluta blanditiis ut, mollitia nisi ex architecto velit, expedita, iure illum quae perferendis? Illum beatae minima quo perferendis vero dolorum.
  </span>
</DocumentFragment>
`;

exports[`Typography should render correctly with variant display-xs 1`] = `
<DocumentFragment>
  <span
    class="typography typography__display-xs typography__medium"
  >
    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Laborum veritatis soluta blanditiis ut, mollitia nisi ex architecto velit, expedita, iure illum quae perferendis? Illum beatae minima quo perferendis vero dolorum.
  </span>
</DocumentFragment>
`;

exports[`Typography should render correctly with variant text-lg 1`] = `
<DocumentFragment>
  <span
    class="typography typography__text-lg typography__medium"
  >
    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Laborum veritatis soluta blanditiis ut, mollitia nisi ex architecto velit, expedita, iure illum quae perferendis? Illum beatae minima quo perferendis vero dolorum.
  </span>
</DocumentFragment>
`;

exports[`Typography should render correctly with variant text-md 1`] = `
<DocumentFragment>
  <span
    class="typography typography__text-md typography__medium"
  >
    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Laborum veritatis soluta blanditiis ut, mollitia nisi ex architecto velit, expedita, iure illum quae perferendis? Illum beatae minima quo perferendis vero dolorum.
  </span>
</DocumentFragment>
`;

exports[`Typography should render correctly with variant text-sm 1`] = `
<DocumentFragment>
  <span
    class="typography typography__text-sm typography__medium"
  >
    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Laborum veritatis soluta blanditiis ut, mollitia nisi ex architecto velit, expedita, iure illum quae perferendis? Illum beatae minima quo perferendis vero dolorum.
  </span>
</DocumentFragment>
`;

exports[`Typography should render correctly with variant text-xl 1`] = `
<DocumentFragment>
  <span
    class="typography typography__text-xl typography__medium"
  >
    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Laborum veritatis soluta blanditiis ut, mollitia nisi ex architecto velit, expedita, iure illum quae perferendis? Illum beatae minima quo perferendis vero dolorum.
  </span>
</DocumentFragment>
`;

exports[`Typography should render correctly with variant text-xs 1`] = `
<DocumentFragment>
  <span
    class="typography typography__text-xs typography__medium"
  >
    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Laborum veritatis soluta blanditiis ut, mollitia nisi ex architecto velit, expedita, iure illum quae perferendis? Illum beatae minima quo perferendis vero dolorum.
  </span>
</DocumentFragment>
`;

exports[`Typography should render correctly with weight bold 1`] = `
<DocumentFragment>
  <span
    class="typography typography__text-md typography__bold"
  >
    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Laborum veritatis soluta blanditiis ut, mollitia nisi ex architecto velit, expedita, iure illum quae perferendis? Illum beatae minima quo perferendis vero dolorum.
  </span>
</DocumentFragment>
`;

exports[`Typography should render correctly with weight medium 1`] = `
<DocumentFragment>
  <span
    class="typography typography__text-md typography__medium"
  >
    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Laborum veritatis soluta blanditiis ut, mollitia nisi ex architecto velit, expedita, iure illum quae perferendis? Illum beatae minima quo perferendis vero dolorum.
  </span>
</DocumentFragment>
`;

exports[`Typography should render correctly with weight regular 1`] = `
<DocumentFragment>
  <span
    class="typography typography__text-md typography__regular"
  >
    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Laborum veritatis soluta blanditiis ut, mollitia nisi ex architecto velit, expedita, iure illum quae perferendis? Illum beatae minima quo perferendis vero dolorum.
  </span>
</DocumentFragment>
`;

exports[`Typography should render correctly with weight semi-bold 1`] = `
<DocumentFragment>
  <span
    class="typography typography__text-md typography__semi-bold"
  >
    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Laborum veritatis soluta blanditiis ut, mollitia nisi ex architecto velit, expedita, iure illum quae perferendis? Illum beatae minima quo perferendis vero dolorum.
  </span>
</DocumentFragment>
`;
